tests/testTransforms.py

```diff
@@ -0,0 +1,468 @@
+#!/usr/bin/env python
+from __future__ import absolute_import, division, print_function
+
+#
+# LSST Data Management System
+# Copyright 2016 LSST/AURA.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import unittest
+import numpy
+import lsst.utils.tests
+import lsst.pex.exceptions
+import lsst.afw.geom
+from lsst.meas.astrom import (
+    PolynomialTransform,
+    ScaledPolynomialTransform,
+    SipForwardTransform,
+    SipReverseTransform,
+    ScaledPolynomialTransformFitter
+)
+
+def makeRandomCoefficientMatrix(n):
+    matrix = numpy.random.randn(n, n)
+    for i in xrange(1, n):
+        matrix[i, (n-i):] = 0;
+    return matrix
+
+def makeRandomAffineTransform():
+    return lsst.afw.geom.AffineTransform(
+        lsst.afw.geom.LinearTransform(numpy.random.randn(2,2)),
+        lsst.afw.geom.Extent2D(*numpy.random.randn(2))
+    )
+
+def makeRandomPolynomialTransform(order, sip=False):
+    xc = makeRandomCoefficientMatrix(order + 1)
+    yc = makeRandomCoefficientMatrix(order + 1)
+    if sip:
+        xc[0,0] = 0
+        yc[0,0] = 0
+        xc[0,1] = 0
+        yc[0,1] = 0
+        xc[1,0] = 0
+        yc[1,0] = 0
+    return PolynomialTransform(xc, yc)
+
+def makeRandomScaledPolynomialTransform(order):
+    return ScaledPolynomialTransform(
+        makeRandomPolynomialTransform(order),
+        makeRandomAffineTransform(),
+        makeRandomAffineTransform()
+    )
+
+def makeRandomSipForwardTransform(order):
+    return SipForwardTransform(
+        lsst.afw.geom.Point2D(*numpy.random.randn(2)),
+        lsst.afw.geom.LinearTransform(numpy.random.randn(2,2)),
+        makeRandomPolynomialTransform(order, sip=True)
+    )
+
+def makeRandomSipReverseTransform(order):
+    origin = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+    cd = lsst.afw.geom.LinearTransform(numpy.random.randn(2,2))
+    poly = makeRandomPolynomialTransform(order, sip=False)
+    return SipReverseTransform(origin, cd, poly)
+
+
```

Comment:
While it is a long file, I'd prefer to keep these together - I don't think there's anything intrinsically bad about having a few hundred lines of code in one file, and the test utility code is just limited to these tests rather than other stuff in meas_astrom.  And, as we've discussed, I expect all of these classes to eventually be replaced by new AST-like things, and if these tests remain valuable then, I expect they'll be moved between packages and it will all be easier to deal with if it's all in one file.

I will run flake8.